### ISSUE TITLE
system_uuid, /srw/www/rmt/public link, Dockerfiles missing directory, missing softlink /var/lib/rmt

### ISSUE THREAD

**Issue by bjin01** [SUSE/rmt#541]
The solution is in general working fine. It took me some time to fix some minor issues:
My environment: SLES15SP1 with docker engine, docker-compose (via pip install docker-compose)

1. in the rmt (container) a file /srv/www/rmt/system_uuid with uuid of the host is mssing. If this is missing then ```rmt-cli sync will return error```. 
Solution: I added system_uuid to the git clone directory on build host.

2. Without a symlink from /var/lib/rmt -> /srv/www/rmt/ I got error return from rmt-cli sync.
Solution: added 
```RUN ln -s /srv/www/rmt /var/lib/rmt``` into Dockerfile

3. Mistake in docker-compose.yml file: the target dir is wrong:
```
volumes:
        - ./public:/srv/www/repo/public/
```
The /srv/www/repo/public/ does not hold the repo subdirs which in turn not shown through nginx.
Solution, change to:
```
volumes:
      - ./public:/srv/www/rmt/public/
```

4. in Dockerfile following dirs need to be created and then ownership changed to _rmt otherwise rmt-cli sync is return permission errors.

Solution in Dockerfile:
```
RUN mkdir -p /srv/www/rmt/public/repo
RUN mkdir -p /srv/www/rmt/public/suma
RUN chown _rmt /srv/www/rmt/public/repo
RUN chown _rmt /srv/www/rmt/public/suma
```
Maybe I'm wrong. But with above changes the docker-compose deployment + successful rmt-cli sync worked for me.

**Comment by bjin01** [comment]
If the 
```/srv/www/rmt/public/repo``` is not set to ownership: _rmt:nginx then I will get below errors:
```
1809ed4cd086:/srv/www/rmt # rmt-cli mirror
I, [2020-04-02T17:12:17.539951 #47]  INFO -- : Mirroring SUSE Manager product tree to /srv/www/rmt/public/suma/
Traceback (most recent call last):
        12: from /srv/www/rmt/lib/rmt/downloader.rb:98:in `block in create_fiber_request'
        11: from /srv/www/rmt/lib/rmt/downloader.rb:186:in `finalize_download'
        10: from /usr/lib64/ruby/2.5.0/fileutils.rb:460:in `mv'
         9: from /usr/lib64/ruby/2.5.0/fileutils.rb:1461:in `fu_each_src_dest'
         8: from /usr/lib64/ruby/2.5.0/fileutils.rb:1479:in `fu_each_src_dest0'
         7: from /usr/lib64/ruby/2.5.0/fileutils.rb:1463:in `block in fu_each_src_dest'
         6: from /usr/lib64/ruby/2.5.0/fileutils.rb:470:in `block in mv'
         5: from /usr/lib64/ruby/2.5.0/fileutils.rb:473:in `rescue in block in mv'
         4: from /usr/lib64/ruby/2.5.0/fileutils.rb:415:in `copy_entry'
         3: from /usr/lib64/ruby/2.5.0/fileutils.rb:1396:in `wrap_traverse'
         2: from /usr/lib64/ruby/2.5.0/fileutils.rb:421:in `block in copy_entry'
         1: from /usr/lib64/ruby/2.5.0/fileutils.rb:1301:in `copy_metadata'
/usr/lib64/ruby/2.5.0/fileutils.rb:1301:in `utime': Operation not permitted @ apply2files - /srv/www/rmt/public/suma/product_tree.json (Errno::EPERM)
1809ed4cd086:/srv/www/rmt # ls -la /srv/www/rmt/public/suma/
total 3112
drwxrwxrwx 2 root root    4096 Apr  2 15:12 .
drwxrwxrwx 5 root  101    4096 Apr  2 14:37 ..
-rwxrwxrwx 1 root root      14 Apr  1 09:58 .gitignore
-rwxrwxrwx 1 root root 3169300 Apr  2 17:12 product_tree.json
1809ed4cd086:/srv/www/rmt # ls -la /srv/www/rmt/public/repo/
total 24
drwxrwxrwx 5 root root 4096 Apr  2 16:30 .
drwxrwxrwx 5 root  101 4096 Apr  2 14:37 ..
-rwxrwxrwx 1 root root   14 Apr  1 09:58 .gitignore
drwxrwxrwx 4 root root 4096 Apr  2 15:12 SUSE
drwxrwxrwx 2 root root 4096 Apr  2 11:05 test1
drwxrwxrwx 2 root root 4096 Apr  2 16:30 testdir
1809ed4cd086:/srv/www/rmt # chown -R _rmt:nginx public/
```

**Comment by tmuntaner** [comment]
Hi @bjin01,

I was just going through the open issues and saw this one. I think that I may have actually solved this in a recent PR #598.

In that PR, I was able to update our [Dockerfile](https://github.com/SUSE/rmt/blob/master/Dockerfile) to use leap 15.2 and now it works with docker-compose and our defined [docker workflow](https://github.com/SUSE/rmt#development-setup---docker-compose).

Thank you for your feedback. If you have any more comments or concerns, please feel free to reopen this issue.
